### ISSUE TITLE
Compiler should emit error when using enum in logical evaluation

### ISSUE THREAD

**Issue by mykohsu** [Microsoft/TypeScript#6754]
Not a great example (random idea borrowed from elsewhere)..."
```
enum E { a, b, c }

let e: E = Math.random() > 0.5 ? E.a : null;
if (e) { // always false
}
```
In Javascript it's common to test a value before use. This can lead to an _unexpected_ case where a defined enum value resolves to false. Sure this makes perfect sense as a `if(0)` but is there any scenario where an enum value should implicitly evaluate to false?

Is there another language where an enum can be directly used in a logic evaluation?

**Comment by RyanCavanaugh** [comment]
It's entirely reasonable to use a value of a bit flag enum like this.

**Comment by mykohsu** [comment]
@RyanCavanaugh do you have an example? In C# you cannot do this without an explicit cast.

**Comment by RyanCavanaugh** [comment]
```ts
enum Flags {
  None = 0,
  Awesome = 1,
  Cool = 2,
  Neat = 4
}
```
let x: Flags = something;
let y: Flags = something;
let z: Flags = x & y;
if(z) { // Equivalent to z === Flags.None
  ...
}

**Comment by mykohsu** [comment]
In this case wouldn't having the default enum start at 1 be safer?

Or is it better in general to:

1. Always declare the first value as None.

2. Test for existence using `if (obj.Flag != null)` instead of `if
(obj.Flag)`
